query-compiler/query-compiler/src/expression/format.rs

```diff
@@ -460,6 +461,34 @@ where
 
         builder.append(self.expression(expr))
     }
+
+    fn extend_record(
+        &'a self,
+        expr: &'a Expression,
+        fields: &'a HashMap<String, RecordValue>,
+    ) -> DocBuilder<'a, PrettyPrinter<'a, D>, ColorSpec> {
+        self.keyword("extend")
+            .append(
+                self.intersperse(
+                    fields.iter().map(|(name, value)| {
+                        let value = match value {
+                            RecordValue::LastInsertId => self.keyword("lastInsertId"),
+                            RecordValue::Value(value) => self.value(value),
+                        };
+
+                        self.field_name(name)
+                            .append(self.text(":"))
+                            .append(self.space())
+                            .append(value)
+                    }),
+                    self.line(),
+                )
+                .append(self.line())
+                .indent(4),
+            )
+            .braces()
+            .append(self.expression(expr))
```

Comment:
Formatting is going to be slightly off here (no whitespace before and after braces; the `extend` keyword is inside the braces and not outside).
```suggestion
        self.keyword("extend")
            .append(self.space())
            .append(
                self.intersperse(
                    fields.iter().map(|(name, value)| {
                        let value = match value {
                            RecordValue::LastInsertId => self.keyword("lastInsertId"),
                            RecordValue::Value(value) => self.value(value),
                        };

                        self.field_name(name)
                            .append(self.text(":"))
                            .append(self.space())
                            .append(value)
                    }),
                    self.line(),
                )
                .append(self.line())
                .indent(4)
                .braces(),
            )
            .append(self.space())
            .append(self.expression(expr))
```

I've also extracted `self.object()` helper on my branch.